algebra/src/main/java/org/almostrealism/collect/CollectionFeatures.java

```diff
@@ -153,21 +279,109 @@ default int size(Supplier s) {
 		}
 	}
 
+	/**
+	 * Gets the total number of elements in a {@link Shape}.
```

Comment:
This is not the total number of elements, but the the number that will be operated on by a single thread of a kernel.